### ISSUE TITLE
Correct installation from PyPI

### ISSUE THREAD

**Issue by mrakitin** [xraypy/xraylarch#231]
Problem:
----
xraylarch package is not available for MacOS in conda channel (`GSECARS` or `GSECARS-mirror`), that's why I'm trying to install xraylarch from PyPI under a newly created conda environment. It installs fine, but when I try to use it, it seems the dependencies from the `requirements.txt` are not installed:
1) Import `larch`:
```python
$ python -c 'import larch'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/mrakitin/anaconda3/envs/larch-pip/lib/python3.6/site-packages/larch/__init__.py", line 10, in <module>
    import numpy
ModuleNotFoundError: No module named 'numpy'
```
I had to manually install the requirements with `pip install -r https://raw.githubusercontent.com/xraypy/xraylarch/master/requirements.txt`, then `pip install xraylarch` worked fine.

2) Then, when I try to import larch, it complains about plugins:
```python
$ python -c 'import larch'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/mrakitin/anaconda3/envs/larch3/lib/python3.6/site-packages/larch/__init__.py", line 32, in <module>
    enable_plugins()
  File "/Users/mrakitin/anaconda3/envs/larch3/lib/python3.6/site-packages/larch/larchlib.py", line 370, in enable_plugins
    import plugins
ModuleNotFoundError: No module named 'plugins'
```
This is observed and tested on both MacOS and Ubuntu 16 under conda environment.

Steps to reproduce:
----
- `conda create -n larch python=3.6`
- `conda activate larch`
- `pip install xraylarch` - should install fine
- `python -c 'import larch'` - should give the error (1)
- `pip install -r https://raw.githubusercontent.com/xraypy/xraylarch/master/requirements.txt` - will solve the error (1)
- `python -c 'import larch'` - should give the error (2)

Expected behavior:
----
`pip install xraylarch` should install the package and all its dependencies successfully on both Linux and MacOS environments, and then should be importable without errors.

Found together with @elistavitski

**Comment by newville** [comment]
@mrakitin Hi, and you are right that we don't yet have a conda package for Py36.  There is good news on this front though, so let me just describe where I think we're at:

-  The git master branch is working pretty well for me with Py36.  I'm trying to use Py36 full time, reverting to Py27 for testing.  There have been a lot of Py3 fixes since the 0.9.35 release.  Most of these were str-vs-bytes (I found and fixed another one of those this morning when trying to write an gzipped Athena project file)  and especially str-vs-bytes for data extracted from HDF5 files.

- We're hoping to release 0.9.36 next week, and will have a conda package for both Py27 and Py36, at least for Linux and MacOS.   We will also update the version on PyPI, but we will continue to encourage "normal install" to use anaconda or even the package created with constructor.

OK, onto your report:

a)  ImportError for `numpy`.   I think this is resolved in the master branch, but I have never tried using pip install with a miniconda environment that didn't have numpy.   I'm not sure I want to test that combination actually.

b) `import larch` raising ModuleNotFound for plugins.  This is fixed in the master branch.  I think it's likely that larch 0.9.35 has enough Python3 errors that have already been fixed in the master branch that it's probably hopeless to try to use the 0.9.35 for Python3.  But installing from the master branch should work.

c) `pip install xraylarch should install the package and all its dependencies`.  Hm, yeah probably.  OTOH, life is short and fully supporting pip, conda, and their mixture is actually work.   If you try to install from the source kit with `python setup.py install`, you'll definitely get feedback about missing modules.

**Comment by newville** [comment]
@mrakitin `pip install xraylarch` now sort-of works for Python 3.6 and 2.7 with 0.9.37.  The source kit includes MacOS binaries.  We do not yet have wheels made or uploaded.

 Using `conda` is still recommended and preferred.

**Comment by mrakitin** [comment]
@newville, thanks for the update and sorry for the delay. I tried to install the package on a clean conda environment using `pip install xraylarch`, but it still complains about the missing packages:
<details>

```bash
$ pip install xraylarch -v
Collecting xraylarch
  1 location(s) to search for versions of xraylarch:
  * https://pypi.python.org/simple/xraylarch/
  Getting page https://pypi.python.org/simple/xraylarch/
  Looking up "https://pypi.python.org/simple/xraylarch/" in the cache
  Current age based on date: 33
  Freshness lifetime from max-age: 600
  Freshness lifetime from request max-age: 600
  The response is "fresh", returning cached response
  600 > 33
  Analyzing links from page https://pypi.python.org/simple/xraylarch/
    Found link https://pypi.python.org/packages/04/2e/293d91f05c40df2b3567c8822309373d6fa54ae451047958f84cfb7a5569/xraylarch-0.9.28.tar.gz#md5=e4a549c18cf7bac41b0f776eba9cae7f (from https://pypi.python.org/simple/xraylarch/), version: 0.9.28
    Found link https://pypi.python.org/packages/54/32/df643f30cc4a96f252c156f0e7e35b546f90f0f1f13187d39205ad0c5abe/xraylarch-0.9.29.tar.gz#md5=ba8f34c9abd373311229121abeedb344 (from https://pypi.python.org/simple/xraylarch/), version: 0.9.29
    Found link https://pypi.python.org/packages/69/95/f20e2b50e4af95780bb71bbf194041e5c95674c874defd08ac013169de58/xraylarch-0.9.30.tar.gz#md5=9c940ec24f83f81fec18cd610bc7c18f (from https://pypi.python.org/simple/xraylarch/), version: 0.9.30
    Found link https://pypi.python.org/packages/57/de/774b6f41c848ea773fdbc1ea0e0334c256c96e693f3c01fedab8c7a2968a/xraylarch-0.9.31.tar.gz#md5=ce7dab319ac1f86752b5498d8533ece2 (from https://pypi.python.org/simple/xraylarch/), version: 0.9.31
    Found link https://pypi.python.org/packages/d0/7c/d7f5fbaa3049502abc783c2efc13a7e4e31943352faf1ef84ae2e499770c/xraylarch-0.9.32.tar.gz#md5=775bfcea11204ef4acd1c8cb3cd75c3e (from https://pypi.python.org/simple/xraylarch/), version: 0.9.32
    Found link https://pypi.python.org/packages/46/d4/7613ff6b20bec7047d8a48669ea88dac48d23d682e15402b9de5a8f69f4a/xraylarch-0.9.33.tar.gz#md5=dd3be483cd253614c7a1ffe32076afc5 (from https://pypi.python.org/simple/xraylarch/), version: 0.9.33
    Found link https://pypi.python.org/packages/6c/ad/c4142a7a0f31bbb2cc8db4844fa9356078b5c3060206940752f0383f1a19/xraylarch-0.9.34.tar.gz#md5=b2b442f306e943d006efeab121eef2c6 (from https://pypi.python.org/simple/xraylarch/), version: 0.9.34
    Found link https://pypi.python.org/packages/40/cd/25592a7d4778a24904dc380d2382a0e41195a770530ee3cdb8aceea2f4b7/xraylarch-0.9.35.tar.gz#md5=bbbf1aa2496d7be633ffbfc63dabe865 (from https://pypi.python.org/simple/xraylarch/), version: 0.9.35
    Found link https://pypi.python.org/packages/39/9d/7c9b35e2acf7e70b26cf2fa27be96bcf844524c48a5c3c1e7d03a652a9fd/xraylarch-0.9.36.tar.gz#md5=46229a06b22e39f210d66e00ac40d069 (from https://pypi.python.org/simple/xraylarch/), version: 0.9.36
    Found link https://pypi.python.org/packages/93/48/2bd92487a7b4a7c0ea1703fab543befa5e60c79a042fd2fbf1e062474f35/xraylarch-0.9.37.tar.gz#md5=28201a80599d27c6dc784689b4b5cdcd (from https://pypi.python.org/simple/xraylarch/), version: 0.9.37
  Using version 0.9.37 (newest of versions: 0.9.28, 0.9.29, 0.9.30, 0.9.31, 0.9.32, 0.9.33, 0.9.34, 0.9.35, 0.9.36, 0.9.37)
  Looking up "https://pypi.python.org/packages/93/48/2bd92487a7b4a7c0ea1703fab543befa5e60c79a042fd2fbf1e062474f35/xraylarch-0.9.37.tar.gz" in the cache
  Current age based on date: 31
  Freshness lifetime from max-age: 31557600
  The response is "fresh", returning cached response
  31557600 > 31
  Using cached xraylarch-0.9.37.tar.gz
  Downloading from URL https://pypi.python.org/packages/93/48/2bd92487a7b4a7c0ea1703fab543befa5e60c79a042fd2fbf1e062474f35/xraylarch-0.9.37.tar.gz#md5=28201a80599d27c6dc784689b4b5cdcd (from https://pypi.python.org/simple/xraylarch/)
  Running setup.py (path:/private/var/folders/nx/0gbmb7qx39709t24sgx368ch0000gn/T/pip-build-92ddfs8x/xraylarch/setup.py) egg_info for package xraylarch
    Running command python setup.py egg_info
    Checking dependencies....

    === Cannot Install Larch, these REQUIRED Modules are missing:
     numpy, scipy, matplotlib, h5py, sqlalchemy, requests, six, psutil, peakutils, asteval, lmfit, termcolor

     Please read INSTALL for further information.

Cleaning up...
No files/directories in /private/var/folders/nx/0gbmb7qx39709t24sgx368ch0000gn/T/pip-build-92ddfs8x/xraylarch/pip-egg-info (from PKG-INFO)
Exception information:
Traceback (most recent call last):
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/pip/commands/install.py", line 335, in run
    wb.build(autobuilding=True)
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/pip/wheel.py", line 756, in build
    self.requirement_set.prepare_files(self.finder)
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/pip/req/req_set.py", line 380, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/pip/req/req_set.py", line 634, in _prepare_file
    abstract_dist.prep_for_dist()
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/pip/req/req_set.py", line 129, in prep_for_dist
    self.req_to_install.run_egg_info()
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/pip/req/req_install.py", line 455, in run_egg_info
    metadata_name = canonicalize_name(self.pkg_info()["Name"])
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/pip/req/req_install.py", line 533, in pkg_info
    data = self.egg_info_data('PKG-INFO')
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/pip/req/req_install.py", line 471, in egg_info_data
    filename = self.egg_info_path(filename)
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/pip/req/req_install.py", line 515, in egg_info_path
    'No files/directories in %s (from %s)' % (base, filename)
```
</details>

After I installed the missing packages manually with `pip install numpy scipy matplotlib h5py sqlalchemy requests six psutil peakutils asteval lmfit termcolor`, the package installed successfully, but the import did not work:
```py
$ python -c 'import larch'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/larch/__init__.py", line 32, in <module>
    enable_plugins()
  File "/Users/mrakitin/anaconda3/envs/xraylarch/lib/python3.6/site-packages/larch/larchlib.py", line 370, in enable_plugins
    import plugins
ModuleNotFoundError: No module named 'plugins'
```

As for the conda installation, it indeed installed successfully in a fresh conda environment using `conda install -yc GSECARS xraylarch`

**Comment by newville** [comment]
@mrakitin Glad that installing with the recommended approach worked!.

It sounds like the not-exactly-supported approach of using `pip install` did not work because some of the site configuration was not set up correctly with `pip install`.   That's not too surprising to me -- the setup/install process is somewhat more involved than the normal procedure with several "post-install steps" being taken, including installing the plugins and making sure they can be found.

The simplest conclusion is that we should be more clear that `python setup.py install` with the source kit *is* supported, but `pip install` is not.   If you really want `pip install` to work, pull requests are welcome.

**Comment by mrakitin** [comment]
Thanks @newville! Please see my attempt to fix the installation with `pip install` in #240.
